lib/sisimai/bite/email/aol.rb

```diff
@@ -0,0 +1,208 @@
+module Sisimai::Bite::Email
+  module Aol
+    # Sisimai::Bite::Email::Aol parses a bounce email which created by Aol Mail.
+    # Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/Aol.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :from    => %r/\APostmaster [<]Postmaster[@]AOL[.]com[>]\z/,
+        :subject => %r/\AUndeliverable: /,
+      }
+      Re1 = {
+        :begin   => %r|\AContent-Type: message/delivery-status|,
+        :rfc822  => %r|\AContent-Type: message/rfc822|,
+        :endof   => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReFailure = {
+        hostunknown: %r/Host[ ]or[ ]domain[ ]name[ ]not[ ]found/,
+        notaccept:   %r/type=MX:[ ]Malformed[ ]or[ ]unexpected[ ]name[ ]server[ ]reply/,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Aol Mail: http://www.aol.com'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+
+      # X-AOL-IP: 192.0.2.135
+      # X-AOL-VSS-INFO: 5600.1067/98281
+      # X-AOL-VSS-CODE: clean
+      # x-aol-sid: 3039ac1afc14546fb98a0945
+      # X-AOL-SCOLL-EIL: 1
+      # x-aol-global-disposition: G
+      # x-aol-sid: 3039ac1afd4d546fb97d75c6
+      # X-BounceIO-Id: 9D38DE46-21BC-4309-83E1-5F0D788EFF1F.1_0
+      # X-Outbound-Mail-Relay-Queue-ID: 07391702BF4DC
+      # X-Outbound-Mail-Relay-Sender: rfc822; shironeko@aol.example.jp
+      def headerlist;  return ['X-AOL-IP']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Aol Mail
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
```

Comment:
__[rubocop]__
* Style/Semicolon: Do not use semicolons to terminate expressions. [:ref](https://github.com/bbatsov/ruby-style-guide#no-semicolon)
* Layout/TrailingWhitespace: Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

<!-- Comment by [SideCI](https://sideci.com) -->